python/ray/experimental/workflow/execution.py

```diff
@@ -64,11 +64,81 @@ def get_output(workflow_id: str) -> ray.ObjectRef:
     """
     assert ray.is_initialized()
     try:
-        actor = ray.get_actor(MANAGEMENT_ACTOR_NAME)
+        workflow_manager = ray.get_actor(MANAGEMENT_ACTOR_NAME)
     except ValueError as e:
         raise ValueError(
             "Failed to connect to the workflow management "
             "actor. The workflow could have already failed. You can use "
             "workflow.resume() to resume the workflow.") from e
-    output = ray.get(actor.get_output.remote(workflow_id))
+    output = ray.get(workflow_manager.get_output.remote(workflow_id))
     return flatten_workflow_output(workflow_id, output)
+
+
+def cancel(workflow_id: str) -> None:
+    try:
+        workflow_manager = ray.get_actor(MANAGEMENT_ACTOR_NAME)
+        ray.get(workflow_manager.cancel_workflow.remote(workflow_id))
+    except ValueError:
+        wf_store = workflow_storage.get_workflow_storage(workflow_id)
+        wf_store.save_workflow_meta(WorkflowMetaData(WorkflowStatus.CANCELED))
+
+
+def get_status(workflow_id: str) -> Optional[WorkflowStatus]:
+    try:
+        workflow_manager = ray.get_actor(MANAGEMENT_ACTOR_NAME)
+        running = ray.get(
+            workflow_manager.is_workflow_running.remote(workflow_id))
+    except Exception:
+        running = False
+    if running:
+        return WorkflowStatus.RUNNING
+    store = workflow_storage.get_workflow_storage(workflow_id)
+    meta = store.load_workflow_meta()
+    if meta is None:
+        return meta
+    return meta.status
+
+
+def list_all(status_filter: Set[WorkflowStatus]) -> Dict[str, WorkflowStatus]:
+    try:
+        workflow_manager = ray.get_actor(MANAGEMENT_ACTOR_NAME)
+    except ValueError:
+        workflow_manager = None
+
+    if workflow_manager is None:
+        runnings = []
+    else:
+        runnings = ray.get(workflow_manager.list_running_workflow.remote())
+    if WorkflowStatus.RUNNING in status_filter and len(status_filter) == 1:
+        return {r: WorkflowStatus.RUNNING for r in runnings}
+
+    runnings = set(runnings)
+    # Here we don't have workflow id, so use empty one instead
+    store = workflow_storage.get_workflow_storage("")
+    ret = {}
+    for (k, s) in store.list_workflow():
```

Comment:
We'll need to change the storage to support listing workflows by status type efficiently. Otherwise this won't scale.